src/components/pages/rules/flyouts/ruleEditor/actionEmailSetup.js

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+import { Trans } from 'react-i18next';
+import dot from 'dot-object';
+
+import { svgs } from 'utilities';
+import { ErrorMsg, Hyperlink, Indicator, Svg } from 'components/shared';
+
+import './actionEmailSetup.css';
+
+export class ActionEmailSetup extends Component {
+  constructor(props) {
+    super(props);
+
+    if (!props.actionSettingsIsPending) {
```

Comment:
Why this check?

Reply:
If the call is already in progress, there is no need to kick it off again.

Reply:
I don't find a reason why it might be pending even before the constructor is kicked off.

Reply:
There is nothing preventing other parts of the app from also calling it. Its just good defensive coding.